subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/espresso/action/TouchOutsideAction.kt

```diff
@@ -0,0 +1,87 @@
+package com.avito.android.test.espresso.action
+
+import android.content.res.Resources
+import android.graphics.Rect
+import android.os.SystemClock
+import android.view.InputDevice
+import android.view.MotionEvent
+import android.view.MotionEvent.PointerProperties
+import android.view.View
+import android.view.ViewConfiguration
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.Configurator
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.Matcher
+
+class TouchOutsideAction : ViewAction {
+
+    private val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    override fun getConstraints(): Matcher<View> = ViewMatchers.isAssignableFrom(View::class.java)
+
+    override fun getDescription(): String = "Press outside of view"
+
+    override fun perform(uiController: UiController, view: View) {
+        val rect = Rect()
+        view.getGlobalVisibleRect(rect)
+
+        //Take left point of our view and add few dp to make sure that point located outside of view
+        val y = uiDevice.displayHeight.toFloat() / 2
+        val x = uiDevice.displayWidth / 2f + rect.width() / 2f + HORIZONTAL_OFFSET_DP.dp()
```

Comment:
So I understand everything now except `dp` as I wrote before you could do two things
1. Use `this` in `dp` function  for multiplying
or
2. Change `dp` to the fun that returns your static `HORIZONTAL_OFFSET_DP` and rename it to `15dp()`